packages/flutter_tools/lib/src/tester/flutter_tester.dart

```diff
@@ -0,0 +1,206 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:convert';
+
+import 'package:meta/meta.dart';
+
+import '../application_package.dart';
+import '../artifacts.dart';
+import '../base/common.dart';
+import '../base/file_system.dart';
+import '../base/io.dart';
+import '../base/process_manager.dart';
+import '../build_info.dart';
+import '../dart/package_map.dart';
+import '../device.dart';
+import '../globals.dart';
+import '../protocol_discovery.dart';
+import '../version.dart';
+
+class FlutterTesterApp extends ApplicationPackage {
+  final String _directory;
+
+  factory FlutterTesterApp.fromCurrentDirectory() {
+    return new FlutterTesterApp._(fs.currentDirectory.path);
+  }
+
+  FlutterTesterApp._(String directory)
+      : _directory = directory,
+        super(id: directory);
+
+  @override
+  String get name => fs.path.basename(_directory);
+
+  @override
+  String get packagePath => fs.path.join(_directory, '.packages');
+}
+
+// TODO(devoncarew): This device does not currently work with full restarts.
+class FlutterTesterDevice extends Device {
+  final _FlutterTesterDeviceLogReader _logReader =
+      new _FlutterTesterDeviceLogReader();
+
+  Process _process;
+
+  FlutterTesterDevice(String deviceId) : super(deviceId);
+
+  @override
+  Future<bool> get isLocalEmulator async => false;
+
+  @override
+  String get name => 'Flutter test device';
+
+  @override
+  DevicePortForwarder get portForwarder => null;
+
+  @override
+  Future<String> get sdkNameAndVersion async {
+    final FlutterVersion flutterVersion = FlutterVersion.instance;
+    return 'Flutter ${flutterVersion.frameworkRevisionShort}';
+  }
+
+  @override
+  Future<TargetPlatform> get targetPlatform async => TargetPlatform.tester;
+
+  @override
+  void clearLogs() {}
+
+  @override
+  DeviceLogReader getLogReader({ApplicationPackage app}) => _logReader;
+
+  @override
+  Future<bool> installApp(ApplicationPackage app) async => true;
+
+  @override
+  Future<bool> isAppInstalled(ApplicationPackage app) async => false;
+
+  @override
+  Future<bool> isLatestBuildInstalled(ApplicationPackage app) async => false;
+
+  @override
+  bool isSupported() => true;
+
+  @override
+  Future<LaunchResult> startApp(ApplicationPackage package,
+      {@required String mainPath,
+      String route,
+      @required DebuggingOptions debuggingOptions,
+      Map<String, dynamic> platformArgs,
+      bool prebuiltApplication: false,
+      bool applicationNeedsRebuild: false,
+      bool usesTerminalUi: true,
+      bool ipv6: false}) async {
```

Comment:
Could you add a trailing comma, for more better formatting?

Reply:
Done.